### ISSUE TITLE
S6: Missing wifi region causes broken cloud connection

### ISSUE THREAD

**Issue by ni-ko-b** [Hypfer/Valetudo#606]
## Describe the bug
No map data is shown

## To Reproduce

1. Go to valetudo in your browser (http://192.168.../)
2. Click on "map"
4. Note how no map is shown. Instead "no map data" appears

### Vacuum Model
Roborock S6

### Valetudo Version
0.6.1

### Firmware Version
1998

### User-Agent
Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/85.0.4183.121 Safari/537.36

## Expected behavior
After docking the robot should display the last created map

## Additional context

- Persistent data has been disabled, map data reset and enabled again

```
[root@rockrobo ~]# cat /mnt/data/valetudo/config.json
{
  "spots": [],
  "mqtt": {
    "enabled": true,
    "server": "*removed*",
    "port": "1883",
    "clientId": "roborock s6",
    "username": "",
    "password": "",
    "usetls": false,
    "ca": "",
    "clientCert": "",
    "clientKey": "",
    "qos": 0,
    "identifier": "rockrobo",
    "topicPrefix": "valetudo",
    "autoconfPrefix": "homeassistant",
    "provideMapData": true
  },
  "webserver": {
    "port": 80
  },
  "dummycloud": {
    "spoofedIP": "203.0.113.1",
    "bindIP": "127.0.0.1"
  },
  "httpAuth": {
    "enabled": false,
    "username": "valetudo",
    "password": "valetudo"
  },
  "allowSSHKeyUpload": true,
  "map_upload_host": "http://127.0.0.1",
  "logLevel": "trace",
  "model": {
    "type": "auto",
    "embedded": true
  },
  "debug": {
    "memoryStatInterval": false
  }
}
```
```

[root@rockrobo ~]# cat /etc/hosts
127.0.0.1       localhost
127.0.1.1       rockrobo

::1             localhost ip6-localhost ip6-loopback
ff02::1         ip6-allnodes
ff02::2         ip6-allrouters

###########################################################
### VALETUDO HOSTS
###########################################################
#
# Add to /etc/hosts
#

203.0.113.1  ot.io.mi.com
2001:db8::1  ot.io.mi.com
203.0.113.1  ott.io.mi.com
2001:db8::1  ott.io.mi.com
203.0.113.1  us.ot.io.mi.com
2001:db8::1  us.ot.io.mi.com
203.0.113.1  us.ott.io.mi.com
2001:db8::1  us.ott.io.mi.com
203.0.113.1  sg.ot.io.mi.com
2001:db8::1  sg.ot.io.mi.com
203.0.113.1  sg.ott.io.mi.com
2001:db8::1  sg.ott.io.mi.com
203.0.113.1  st.ot.io.mi.com
2001:db8::1  st.ot.io.mi.com
203.0.113.1  st.ott.io.mi.com
2001:db8::1  st.ott.io.mi.com
203.0.113.1  de.ot.io.mi.com
2001:db8::1  de.ot.io.mi.com
203.0.113.1  de.ott.io.mi.com
2001:db8::1  de.ott.io.mi.com
203.0.113.1  ru.ot.io.mi.com
2001:db8::1  ru.ot.io.mi.com
203.0.113.1  ru.ott.io.mi.com
2001:db8::1  ru.ott.io.mi.com
203.0.113.1  in.ot.io.mi.com
2001:db8::1  in.ot.io.mi.com
203.0.113.1  in.ott.io.mi.com
2001:db8::1  in.ott.io.mi.com
203.0.113.1  tw.ot.io.mi.com
2001:db8::1  tw.ot.io.mi.com
203.0.113.1  tw.ott.io.mi.com
2001:db8::1  tw.ott.io.mi.com
203.0.113.1  ea.ot.io.mi.com
2001:db8::1  ea.ot.io.mi.com
203.0.113.1  ea.ott.io.mi.com
2001:db8::1  ea.ott.io.mi.com
203.0.113.1  pv.ot.io.mi.com
2001:db8::1  pv.ot.io.mi.com
203.0.113.1  pv.ott.io.mi.com
2001:db8::1  pv.ott.io.mi.com
203.0.113.1  cdn.cnbj0.files.fds.api.xiaomi.com
2001:db8::1  cdn.cnbj0.files.fds.api.xiaomi.com
203.0.113.1  cnbj0.fds.api.xiaomi.com
2001:db8::1  cnbj0.fds.api.xiaomi.com
203.0.113.1  cnbj2.fds.api.xiaomi.com
2001:db8::1  cnbj2.fds.api.xiaomi.com
203.0.113.1  awsbj0.fds.api.xiaomi.com
2001:db8::1  awsbj0.fds.api.xiaomi.com
203.0.113.1  awsbj0-files.fds.api.xiaomi.com
2001:db8::1  awsbj0-files.fds.api.xiaomi.com
203.0.113.1  awsusor0.fds.api.xiaomi.com
2001:db8::1  awsusor0.fds.api.xiaomi.com
203.0.113.1  awsusor0-files.fds.api.xiaomi.com
2001:db8::1  awsusor0-files.fds.api.xiaomi.com
203.0.113.1  awssgp0.fds.api.xiaomi.com
2001:db8::1  awssgp0.fds.api.xiaomi.com
203.0.113.1  awssgp0-files.fds.api.xiaomi.com
2001:db8::1  awssgp0-files.fds.api.xiaomi.com
203.0.113.1  awsde0.fds.api.xiaomi.com
2001:db8::1  awsde0.fds.api.xiaomi.com
203.0.113.1  awsde0-files.fds.api.xiaomi.com
2001:db8::1  awsde0-files.fds.api.xiaomi.com
203.0.113.1  ksyru0-eco.fds.api.xiaomi.com
2001:db8::1  ksyru0-eco.fds.api.xiaomi.com
203.0.113.1  ksyru0-eco-files.fds.api.xiaomi.com
2001:db8::1  ksyru0-eco-files.fds.api.xiaomi.com
203.0.113.1  awsind0-eco.fds.api.xiaomi.com
2001:db8::1  awsind0-eco.fds.api.xiaomi.com
203.0.113.1  awsind0-eco-files.fds.api.xiaomi.com
2001:db8::1  awsind0-eco-files.fds.api.xiaomi.com
203.0.113.1  iot.roborock.com
2001:db8::1  iot.roborock.com
```

```
[root@rockrobo ~]# cat /etc/rc.local
#!/bin/sh -e
#
# rc.local
#
# This script is executed at the end of each multiuser runlevel.
# Make sure that the script will "" on success or any other
# value on error.
#
# In order to enable or disable this script just change the execution
# bits.
#
# By default this script does nothing.


###########################################################
### VALETUDO RC.LOCAL
###########################################################
#
# Add to /etc/rc.local before the "exit 0" command

iptables         -F OUTPUT
ip6tables        -F OUTPUT
iptables  -t nat -F OUTPUT
iptables  -t nat -A OUTPUT -p tcp --dport 80   -d 203.0.113.1 -j DNAT --to-destination 127.0.0.1:8053
iptables  -t nat -A OUTPUT -p udp --dport 8053 -d 203.0.113.1 -j DNAT --to-destination 127.0.0.1:8053
iptables         -A OUTPUT                     -d 203.0.113.1/32  -j REJECT
ip6tables        -A OUTPUT                     -d 2001:db8::1/128 -j REJECT

ip route add 203.0.113.1 via 127.0.0.1
### VALETUDO RC.LOCAL EXIT ###
exit 0
```

```
[root@rockrobo ~]# iptables -vL -t filter
Chain INPUT (policy ACCEPT 8590 packets, 1652K bytes)
 pkts bytes target     prot opt in     out     source               destination
    0     0 DROP       tcp  --  any    any     anywhere             anywhere             tcp dpt:29
    0     0 DROP       tcp  --  any    any     anywhere             anywhere             tcp dpt:29

Chain FORWARD (policy ACCEPT 0 packets, 0 bytes)
 pkts bytes target     prot opt in     out     source               destination

Chain OUTPUT (policy ACCEPT 7869 packets, 1667K bytes)
 pkts bytes target     prot opt in     out     source               destination
    0     0 REJECT     all  --  any    any     anywhere             ot.io.mi.com         reject-with icmp-port-unreachable
```

```
[root@rockrobo ~]# iptables -vL -t nat
Chain PREROUTING (policy ACCEPT 295 packets, 63802 bytes)
 pkts bytes target     prot opt in     out     source               destination

Chain INPUT (policy ACCEPT 167 packets, 38939 bytes)
 pkts bytes target     prot opt in     out     source               destination

Chain OUTPUT (policy ACCEPT 28 packets, 1951 bytes)
 pkts bytes target     prot opt in     out     source               destination
    0     0 DNAT       tcp  --  any    any     anywhere             ot.io.mi.com         tcp dpt:www to:127.0.0.1:8053
    0     0 DNAT       udp  --  any    any     anywhere             ot.io.mi.com         udp dpt:8053 to:127.0.0.1:8053

Chain POSTROUTING (policy ACCEPT 28 packets, 1951 bytes)
 pkts bytes target     prot opt in     out     source               destination
```

```
[root@rockrobo ~]# cat /mnt/data/rockrobo/rrlog/rrlog.log
2980067 rrlogd T488 compress file rrlog.20200927171707599.log, then delete it
2980072 rrlogd T488 nice -n 19 ionice -c2 -n3 xz -0 -c /mnt/data/rockrobo/rrlog/000108.20200927171707443_R0314S01400694_2020060901REL/rrlog.20200927171707599.log > /mnt/data/rockrobo/rrlog/cf9c5d20.tmp
2980114 rrlogd T456 ==================catch SIGUSR3--STAT  end ===================
2980132 rrlogd T456 Europe/Berlin -> 19:17:08 offset:-1
2980132 rrlogd T456 notify upload thread to work ondock = 1
2980132 rrlogd T456 2979418 librrlog open tmp /dev/shm/SLAM_fprintf.tmp.log

2980133 rrlogd T482 select_fds_host:
2980133 rrlogd T482 cannot get log server, abort upload
2980138 rrlogd T488 run nice -n 19 ionice -c2 -n3 xz -0 -c /mnt/data/rockrobo/rrlog/000108.20200927171707443_R0314S01400694_2020060901REL/rrlog.20200927171707599.log > /mnt/data/rockrobo/rrlog/cf9c5d20.tmp successfully.
2980165 rrlogd T488 mv: /mnt/data/rockrobo/rrlog/cf9c5d20.tmp.enc to /mnt/data/rockrobo/rrlog/000108.20200927171707443_R0314S01400694_2020060901REL/rrlog.20200927171707599.log.rr2.xz
2980165 rrlogd T488 rename: /mnt/data/rockrobo/rrlog/cf9c5d20.tmp.enc=>/mnt/data/rockrobo/rrlog/000108.20200927171707443_R0314S01400694_2020060901REL/rrlog.20200927171707599.log.rr2.xz
2980165 rrlogd T488 rm: /mnt/data/rockrobo/rrlog/cf9c5d20.tmp
2980166 rrlogd T488 /mnt/data/rockrobo/rrlog/000108.20200927171707443_R0314S01400694_2020060901REL/rrlog.20200927171707599.log done
2980166 rrlogd T488 rm: /mnt/data/rockrobo/rrlog/000108.20200927171707443_R0314S01400694_2020060901REL/rrlog.20200927171707599.log
2980167 rrlogd T488 pl_name: varlog.tar.rr2.xz to varlog.tar.pl3.rr2.xz
2980167 rrlogd T488 mv: /mnt/data/rockrobo/rrlog/000108.20200927171707443_R0314S01400694_2020060901REL/varlog.tar.rr2.xz to /mnt/data/rockrobo/rrlog/000108.20200927171707443_R0314S01400694_2020060901REL/varlog.tar.pl3.rr2.xz
2980168 rrlogd T488 pl_name: miio.rr2.xz to miio.pl9.rr2.xz
2980168 rrlogd T488 mv: /mnt/data/rockrobo/rrlog/000108.20200927171707443_R0314S01400694_2020060901REL/miio.rr2.xz to /mnt/data/rockrobo/rrlog/000108.20200927171707443_R0314S01400694_2020060901REL/miio.pl9.rr2.xz
2980168 rrlogd T488 pl_name: SLAM_fprintf.log.0001.rr2.xz to SLAM_fprintf.log.0001.pl9.rr2.xz
2980168 rrlogd T488 mv: /mnt/data/rockrobo/rrlog/000108.20200927171707443_R0314S01400694_2020060901REL/SLAM_fprintf.log.0001.rr2.xz to /mnt/data/rockrobo/rrlog/000108.20200927171707443_R0314S01400694_2020060901REL/SLAM_fprintf.log.0001.pl9.rr2.xz
2980168 rrlogd T488 pl_name: topH.last.log.xz to topH.last.log.pl3.xz
2980168 rrlogd T488 mv: /mnt/data/rockrobo/rrlog/000108.20200927171707443_R0314S01400694_2020060901REL/topH.last.log.xz to /mnt/data/rockrobo/rrlog/000108.20200927171707443_R0314S01400694_2020060901REL/topH.last.log.pl3.xz
2980169 rrlogd T488 pl_name: top.last.log.xz to top.last.log.pl3.xz
2980169 rrlogd T488 mv: /mnt/data/rockrobo/rrlog/000108.20200927171707443_R0314S01400694_2020060901REL/top.last.log.xz to /mnt/data/rockrobo/rrlog/000108.20200927171707443_R0314S01400694_2020060901REL/top.last.log.pl3.xz
2980169 rrlogd T488 pl_name: rrlog.20200927171707599.log.rr2.xz to rrlog.20200927171707599.log.pl6.rr2.xz
2980169 rrlogd T488 mv: /mnt/data/rockrobo/rrlog/000108.20200927171707443_R0314S01400694_2020060901REL/rrlog.20200927171707599.log.rr2.xz to /mnt/data/rockrobo/rrlog/000108.20200927171707443_R0314S01400694_2020060901REL/rrlog.20200927171707599.log.pl6.rr2.xz
2980170 rrlogd T488 pl_name: 000108_1601224047087_v1.rr.stat to 000108_1601224047087_v1.pl9.rr.stat
2980170 rrlogd T488 mv: /mnt/data/rockrobo/rrlog/000108.20200927171707443_R0314S01400694_2020060901REL/000108_1601224047087_v1.rr.stat to /mnt/data/rockrobo/rrlog/000108.20200927171707443_R0314S01400694_2020060901REL/000108_1601224047087_v1.pl9.rr.stat
2980170 rrlogd T488 pl_name: watchdog.rr2.xz to watchdog.pl3.rr2.xz
2980170 rrlogd T488 mv: /mnt/data/rockrobo/rrlog/000108.20200927171707443_R0314S01400694_2020060901REL/watchdog.rr2.xz to /mnt/data/rockrobo/rrlog/000108.20200927171707443_R0314S01400694_2020060901REL/watchdog.pl3.rr2.xz
2980170 rrlogd T488 pl_name: boot_reason.rr2.xz to boot_reason.pl9.rr2.xz
2980170 rrlogd T488 mv: /mnt/data/rockrobo/rrlog/000108.20200927171707443_R0314S01400694_2020060901REL/boot_reason.rr2.xz to /mnt/data/rockrobo/rrlog/000108.20200927171707443_R0314S01400694_2020060901REL/boot_reason.pl9.rr2.xz
2980171 rrlogd T488 tar_match_file top.last.log.pl3.xz
2980172 rrlogd T488 tar_match_file SLAM_fprintf.log.0001.pl9.rr2.xz
2980172 rrlogd T488 tar_match_file topH.last.log.pl3.xz
2980172 rrlogd T488 nice -n 19 ionice -c2 -n3 tar cvf top.tar top.last.log.pl3.xz
top.last.log.pl3.xz
2980196 rrlogd T488 run nice -n 19 ionice -c2 -n3 tar cvf top.tar top.last.log.pl3.xz successfully.
2980196 rrlogd T488 rm: top.last.log.pl3.xz
2980197 rrlogd T488 nice -n 19 ionice -c2 -n3 tar cvf SLAM_fprintf.tar SLAM_fprintf.log.0001.pl9.rr2.xz
SLAM_fprintf.log.0001.pl9.rr2.xz
2980222 rrlogd T488 run nice -n 19 ionice -c2 -n3 tar cvf SLAM_fprintf.tar SLAM_fprintf.log.0001.pl9.rr2.xz successfully.
2980222 rrlogd T488 rm: SLAM_fprintf.log.0001.pl9.rr2.xz
2980222 rrlogd T488 nice -n 19 ionice -c2 -n3 tar rvf top.tar topH.last.log.pl3.xz
topH.last.log.pl3.xz
2980246 rrlogd T488 run nice -n 19 ionice -c2 -n3 tar rvf top.tar topH.last.log.pl3.xz successfully.
2980246 rrlogd T488 rm: topH.last.log.pl3.xz
2980263 rrlogd T488 Europe/Berlin -> 19:17:08 offset:-1
2980263 rrlogd T488 notify upload thread to work ondock = 1
2980264 rrlogd T482 select_fds_host:
2980264 rrlogd T482 cannot get log server, abort upload
2980264 rrlogd T488 found /mnt/data/rockrobo/rrlog/000107.20200927163502106_R0314S01400694_2020060901REL/APP_PROXY_normal.tar. will not delete
2980264 rrlogd T488 found /mnt/data/rockrobo/rrlog/000106.20200927161208622_R0314S01400694_2020060901REL/watchdog.pl3.rr2.xz. will not delete
2980264 rrlogd T488 found /mnt/data/rockrobo/rrlog/000105.20200927161139850_R0314S01400694_2020060901REL/APP_PROXY_normal.tar. will not delete
2980265 rrlogd T488 found /mnt/data/rockrobo/rrlog/000104.20200927152555588_R0314S01400694_2020060901REL/watchdog.pl3.rr2.xz. will not delete
2980265 rrlogd T488 found /mnt/data/rockrobo/rrlog/000103.20200927152426314_R0314S01400694_2020060901REL/APP_PROXY_normal.tar. will not delete
2980266 rrlogd T488 found /mnt/data/rockrobo/rrlog/000102.20200927142335387_R0314S01400694_2020060901REL/APP_PROXY_normal.tar. will not delete
2980266 rrlogd T488 found /mnt/data/rockrobo/rrlog/000101.20200927125259843_R0314S01400694_2020060901REL/watchdog.pl3.rr2.xz. will not delete
2980266 rrlogd T488 found /mnt/data/rockrobo/rrlog/000100.20200927111254543_R0314S01400694_2020060901REL/000100_1601168707989_v1.pl9.rr.stat. will not delete
2980267 rrlogd T488 found /mnt/data/rockrobo/rrlog/000099.20200927100254751_R0314S01400694_2020060901REL/APP_PROXY_normal.tar. will not delete
2980267 rrlogd T488 found /mnt/data/rockrobo/rrlog/000098.20200927010456565_R0314S01400694_2020060901REL/watchdog.pl3.rr2.xz. will not delete
2980267 rrlogd T488 found /mnt/data/rockrobo/rrlog/000097.20200926154014711_R0314S01400694_2020060901REL/watchdog.pl3.rr2.xz. will not delete
2980268 rrlogd T488 found /mnt/data/rockrobo/rrlog/000096.20200926153131587_R0314S01400694_2020060901REL/APP_PROXY_normal.tar. will not delete
2980269 rrlogd T488 found /mnt/data/rockrobo/rrlog/000108.20200927171707443_R0314S01400694_2020060901REL/watchdog.pl3.rr2.xz. will not delete
2980270 rrlogd T488 found /mnt/data/rockrobo/rrlog/000107.20200927163502106_R0314S01400694_2020060901REL/APP_PROXY_normal.tar. will not delete
2980270 rrlogd T488 found /mnt/data/rockrobo/rrlog/000106.20200927161208622_R0314S01400694_2020060901REL/watchdog.pl3.rr2.xz. will not delete
2980271 rrlogd T488 found /mnt/data/rockrobo/rrlog/000105.20200927161139850_R0314S01400694_2020060901REL/APP_PROXY_normal.tar. will not delete
2980271 rrlogd T488 found /mnt/data/rockrobo/rrlog/000104.20200927152555588_R0314S01400694_2020060901REL/watchdog.pl3.rr2.xz. will not delete
2980271 rrlogd T488 found /mnt/data/rockrobo/rrlog/000103.20200927152426314_R0314S01400694_2020060901REL/APP_PROXY_normal.tar. will not delete
2980272 rrlogd T488 found /mnt/data/rockrobo/rrlog/000102.20200927142335387_R0314S01400694_2020060901REL/APP_PROXY_normal.tar. will not delete
2980272 rrlogd T488 found /mnt/data/rockrobo/rrlog/000101.20200927125259843_R0314S01400694_2020060901REL/watchdog.pl3.rr2.xz. will not delete
2980272 rrlogd T488 found /mnt/data/rockrobo/rrlog/000100.20200927111254543_R0314S01400694_2020060901REL/000100_1601168707989_v1.pl9.rr.stat. will not delete
2980273 rrlogd T488 found /mnt/data/rockrobo/rrlog/000099.20200927100254751_R0314S01400694_2020060901REL/APP_PROXY_normal.tar. will not delete
2980273 rrlogd T488 found /mnt/data/rockrobo/rrlog/000098.20200927010456565_R0314S01400694_2020060901REL/watchdog.pl3.rr2.xz. will not delete
2980274 rrlogd T488 found /mnt/data/rockrobo/rrlog/000097.20200926154014711_R0314S01400694_2020060901REL/watchdog.pl3.rr2.xz. will not delete
2980274 rrlogd T488 found /mnt/data/rockrobo/rrlog/000096.20200926153131587_R0314S01400694_2020060901REL/APP_PROXY_normal.tar. will not delete
3007515 rrlogd T456 fl_notify_self: msglen=4, offset=4, ret=4, sig=4
```

- It seems like the robot does not upload the map data to valetudo ("cannot get log server, abort upload"). Also no attempts by valetudo itself seem to be logged. I assume there is some issue with the iptables configuration.

**Comment by ni-ko-b** [comment]
[valetudo.log](https://github.com/Hypfer/Valetudo/files/5288757/valetudo.log)

**Comment by Hypfer** [comment]
Please try this solution

```
Dgi, [27.08.20 21:58]
[Forwarded from Dgi]
#!/bin/bash

echo ssid=\"MySSID\" > /mnt/data/miio/wifi.conf
echo psk=\"MyPassword\" >> /mnt/data/miio/wifi.conf
echo key_mgmt=\"WPA\" >> /mnt/data/miio/wifi.conf
echo uid=0 >> /mnt/data/miio/wifi.conf
echo region=us >> /mnt/data/miio/wifi.conf
echo cfg_by=miot >> /mnt/data/miio/wifi.conf
echo 0 > /mnt/data/miio/device.uid
echo "us" > /mnt/data/miio/device.country

Dgi, [27.08.20 21:59]
(replace ssid, password and region, if needed)
```

**Comment by ni-ko-b** [comment]
Sorry I forgot to mention that the robot was already provisioned:

```
[root@rockrobo ~]# cat /mnt/data/miio/wifi.conf
ssid="*removed*"
psk="*removed*"
key_mgmt=WPA
uid=0
region=
cfg_by=miot
```

**Comment by Hypfer** [comment]
Try adding the region then. Afaik the missing value there is causing the miio client to not connect at all

**Comment by ni-ko-b** [comment]
ok thank you. I just patched the files and will report back

**Comment by ni-ko-b** [comment]
it works now. thank you very much for the incredible fast support and this awesome software :) is it possible to donate?

**Comment by Hypfer** [comment]
The best way to donate is to donate your time by supporting other users :)

This missing region issue is indeed a bug btw. Valetudo is missing something in the provisioning command

**Comment by ni-ko-b** [comment]
Awesome. Glad I could help identify this :D

**Comment by Hypfer** [comment]
Should be fixed with 28483abd4431cf0d38953aefeb3995e9d45ea2b8
